lib/utilities/sign.js

```diff
@@ -58,20 +155,85 @@ async function hardwareSignatures(hwApp, ptx, account, enc) {
  * @param {[]String}   paths  - return signature in hex
  */
 async function hardwareTransaction(hwApp, mtx, inputs, paths) {
-  const ledgerInputs = [];
-  for (const [i, input] of mtx.inputs.entries()) {
-    const li = new LedgerTXInput({
-      tx: inputs[i].toRaw(),
-      index: input.prevout.index,
-      path: paths[i],
-      witness: inputs[i].hasWitness(),
-    });
-    ledgerInputs.push(li);
-  }
+  switch (hwApp.type) {
+    case 'ledger': {
+      const ledgerInputs = [];
+      for (const [i, input] of mtx.inputs.entries()) {
+        const li = new LedgerTXInput({
+          tx: inputs[i].toRaw(),
+          index: input.prevout.index,
+          path: paths[i],
+          witness: inputs[i].hasWitness(),
+        });
+        ledgerInputs.push(li);
+      }
+
+      await hwApp.signTransaction(mtx, ledgerInputs);
+
+      return mtx;
+    }
+    case 'trezor': {
+      // TrezorConnect uses their backend to fetch
+      // transaction data, it doesn't "just sign"
+      // Build data structure for trezor.js, which
+      // will be deprecated at some point in the future
+      // TODO: determine if TrezorConnect works
+      // with SPENDMULTISIG
+
+      // data structures for session.signTX
+
+      const trezorInputs = [];
 
-  await hwApp.signTransaction(mtx, ledgerInputs);
+      for (const [i, input] of mtx.inputs.entries()) {
+        const path = parsePath(paths[i], true);
+        let ti = {
+          address_n: path,
+          prev_hash: input.prevout.hash.toString('hex'),
+          prev_index: input.prevout.index,
+          sequence: input.sequence,
+          // default to normal spend
+          script_type: 'SPENDADDRESS',
```

Comment:
See https://github.com/trezor/trezor.js/blob/master/src/utils/signbjstx.js#L42

Copy this when segwit input